### ISSUE TITLE
`no-array-from-fill`: False positive on canvas context .fill()

### ISSUE THREAD

**Issue by Syzgyn** [sindresorhus/eslint-plugin-unicorn#3632]
`unicorn/no-array-from-fill` checks `fill` methods that do not belong to an array.

For example, [`CanvasRenderingContext2D`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D) has a `fill()` method that triggers the `no-array-from-fill` rule when passing a [`Path2D`](https://developer.mozilla.org/en-US/docs/Web/API/Path2D) object to it.

```js
function drawSliceTs(ctx: CanvasRenderingContext2D) {
  const slice = new Path2D()
  ctx.fill(slice) // error:  Do not use a reference value as the fill value
```

Repo with reproducible error: https://github.com/Syzgyn/unicorn-array-fill

**Comment by koding88** [comment]
I investigated this — the false positive occurs because `CanvasRenderingContext2D` and `OffscreenCanvasRenderingContext2D` are not recognized in `is-array.js`'s `knownNonArrayTypeNames`.

I've drafted a fix with test coverage on `koding88/eslint-plugin-unicorn` (branch `fix/no-array-fill-canvas-context`). Happy to open a PR if external contributions are welcomed for this issue.

**Comment by Syzgyn** [comment]
Thank you!
